online doc: fix links, spelling and formatting

slawr opened the pull request:
Fix some miscellaneous errors related to broken links, spelling and formatting in the online doc. See commit msgs for details.

Did some manual render testing and the changes seem to work. Whilst setting up for that I spotted a documentation gap around the Hugo versions for which I will raise an separate issue.

SebastianSchildt commented on docs-gen/content/rule_set/data_entry/_index.md:
````diff
@@ -11,7 +11,7 @@ In order to help application developers, who are using the specification, it mak
 a publisher (or producer) that continuously updates the signal value while an
 attribute has a set value, defined in the specification, that never changes.
-As summary, besides [```branch```](/rule_set/branches) type can be:
+As summary, besides [```branch```](/vehicle_signal_specification/rule_set/branches) type can be:
````
[```branch```] is linked here (correctly), however in the same paragraph frither up sensor/actuator and stream (is this ever used?) are not.

slawr replied:
I was deliberately constraining myself to existing links that were broken in this way, but yes I can push some additional links. What is the correct path for stream?

slawr replied:
Added links in PR update.

SebastianSchildt commented on docs-gen/content/tools/usage.md:
```diff
@@ -4,7 +4,7 @@ date: 2019-08-04T13:21:34+02:00
 ---
 After installing the dependencies, as described in the
-[requirements](/tools/requirements), you can simply run:
+[requirements](/vehicle_signal_specification/tools/requirements), you can simply run:
```
The link is fixed, but that documentations is outdated, I would suggest removing the requirments doc (or all tool docs) completely and refer to the vss-tools github

slawr replied:
Yes, I noticed consistencies in the online tools doc and the READMEs that I alluded to in https://github.com/GENIVI/vss-tools/issues/74. It's worth noting that the online doc containing information not in the README. Seeing rationalising the doc as a related but bigger task I stuck to fixing the broken links. I would suggest moving your good input into an enhancement ticket.

slawr replied:
It would have been nice if we had simply merged this simple fix first before your new PR to completely rework the tools documentation was subsequently pushed and merged. It would have saved me having to pick apart my commit and rebase for what is a straight forward PR.

Force pushed an update with the change dropped as it no longer exists upstream.

SebastianSchildt reviewed (approved):
Except my comments, approve, as it _does_ fix links and spellings (as promised) and everything seems to work on a local hugo. Merging this will make things better. But there is always more to do...

gunnarx commented:
Let's keep it open and add links to sensor/actuator/stream as @SebastianSchildt proposed.

slawr commented:
@gunnarx once someone has pointed me at the path for the stream link I'll send an update for the additional links asked for by @SebastianSchildt

slawr commented:
@gunnarx @SebastianSchildt any feedback on what is the correct url for the stream node type? I can find no example in the online doc. The nearest related text seems to be https://genivi.github.io/vehicle_signal_specification/rule_set/data_entry/data_types/#data-streams.

I would like to get this PR resolved but currently it is difficult to fulfill Sebastian's additional requests.

SebastianSchildt commented:
@slawr   I think link is fine/close enough for the documentation.

Streams are one of those things defined, but never used in the standard catalogue.

slawr commented:
Updated:

1. Added new links requested in @SebastianSchildt review now needed link agreed.
2. Rebased and dropped tool doc from this PR due to https://github.com/GENIVI/vehicle_signal_specification/pull/311 that was pushed and merged whilst this was open.

Should now be ready for final review or merge.
